### PR TITLE
ref(monitors): Use `monitor_config` for upsert during checkin

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This is a bit more cleary while being slightly more verbose

**NOTE:** This was NOT yet being used in production. It is a breaking change, but luckily no one was using it :-)

### CODE REVIEW

File: src/sentry/monitors/validators.py [hunk 1]
```diff
@@ -179,16 +179,17 @@ class MonitorCheckInValidator(serializers.Serializer):
...
     def validate(self, attrs):
         attrs = super().validate(attrs)
 
         # Support specifying monitor configuration via a check-in
         #
-        # NOTE: Most monitor attributes are contextual, the monitor config is
-        #       passed in via this checkin searializers config attribute.
-        monitor_config = attrs.get("config")
+        # NOTE: Most monitor attributes are contextual (project, slug, etc),
+        #       the monitor config is passed in via this checkin searializers
```
Comment (rjo100) [inline src/sentry/monitors/validators.py:190]: `serializer's`

File: src/sentry/monitors/validators.py [hunk 2]
```diff
@@ -207,7 +208,10 @@ def validate(self, attrs):
                 },
             )
             monitor_validator.is_valid(raise_exception=True)
+
+            # Drop the `monitor_config` attribute favor in favor of the fully
+            # validate monitor data
```
Comment (rjo100) [inline src/sentry/monitors/validators.py:213]: `validated`

**Review by rjo100 — APPROVED** [review]

File: src/sentry/monitors/validators.py (hunk 1 above)
Comment (evanpurkhiser) [inline src/sentry/monitors/validators.py:190]:
```suggestion
        #       the monitor config is passed in via this checkin serializer's
```

File: src/sentry/monitors/validators.py (hunk 2 above)
Comment (evanpurkhiser) [inline src/sentry/monitors/validators.py:213]:
```suggestion
            # validated monitor data
```

**Pull request merged**